elephant/spike_train_generation.py

```diff
@@ -516,13 +513,9 @@ def _thinning_for_refractory_period(spiketrain, refractory_period):
     thinned_spiketrain : np.ndarray
         thinned out spiketrain
     """
-    thinned_spiketrain = []
-    previous_spike = -refractory_period
-    for spike in spiketrain:
-        if spike > previous_spike + refractory_period:
-            thinned_spiketrain.append(spike)
-            previous_spike = spike
-    return np.array(thinned_spiketrain)
+    mask_valid = np.diff(spiketrain, prepend=0) > refractory_period
```

Comment:
It is this mask_valid, that does not work. I mean the for loop does not look nice, I know. But it can provide, compare only with the spikes that are actually taken. 
But now, let's say, e.g. the firing rate is very high, at the spiketrain every ISI is smaller than the refractory period. In this mask, say would be all False. And thus all spikes would be discarded...
So I would return to the old version until a very great idea arises on how to do it with numpy.